mono_repo/lib/src/commands/travis.dart

```diff
@@ -323,19 +324,103 @@ List<Object> _calculateOrderedStages(RootConfig rootConfig) {
 /// Lists all the jobs, setting their stage, environment, and script.
 Iterable<Map<String, String>> _listJobs(
     Iterable<TravisJob> jobs, Map<String, String> commandsToKeys) sync* {
+  final jobEntries = <_TravisJobEntry>[];
+
   for (var job in jobs) {
     final commands =
-        job.tasks.map((task) => commandsToKeys[task.command]).join(' ');
-    final jobName = 'SDK: ${job.sdk}; '
-        'PKGS: ${job.package}; '
+        job.tasks.map((task) => commandsToKeys[task.command]).toList();
+
+    jobEntries.add(_TravisJobEntry(job, commands));
+  }
+
+  final groupedItems =
+      groupBy<_TravisJobEntry, _TravisJobEntry>(jobEntries, (e) => e);
+
+  for (var entry in groupedItems.entries) {
+    final mergedJobEntries =
+        entry.value.where((je) => je.job.mergeJobs).toList();
+    final packages = mergedJobEntries.map((t) => t.job.package).toList();
+    final unmergedJobEntries =
+        entry.value.where((je) => !je.job.mergeJobs).toList();
+
+    final jobDetails = {
+      'Dart SDK': entry.key.job.sdk,
+      'Stage   ': entry.key.job.stageName,
+      'Task    ': entry.key.job.name,
+    }.entries.map((e) => '  ${e.key}: ${e.value}').join('\n');
+
+    if (mergedJobEntries.isNotEmpty) {
+      if (mergedJobEntries.length == 1) {
+        final firstJob = mergedJobEntries.single.job;
+        print('');
+        print(yellow.wrap('${firstJob.package}: there is a group with '
```

Comment:
I don't know if this is actually worth warning about - you might still want it to be merged if it can be later on.

Reply:
I went a very different route – put `merge_stages` in `mono_repo.yaml` WAY cleaner.

See updates.